pkg/controllers/job/job_controller.go

```diff
@@ -239,6 +242,11 @@ func (cc *Controller) processNextReq() bool {
 	glog.V(3).Infof("Execute <%v> on Job <%s/%s> in <%s> by <%T>.",
 		action, req.Namespace, req.JobName, jobInfo.Job.Status.State.Phase, st)
 
+	if action != vkbatchv1.SyncJobAction {
+		cc.recordJobEvent(jobInfo.Job.Namespace, jobInfo.Job.Name, vkbatchv1.ExecuteAction, fmt.Sprintf(
```

Comment:
That's ok to record event for `SyncJobAction`.

Reply:
If the syncAction event is included it generates a lot of events. If included it is observed that the spamfilter of EventBroadcaster then blocks the similar events on that job object. Because of which new events like restartjob/completejob events were not getting send to the api server.